subsys/sip_svc/Kconfig

```diff
@@ -0,0 +1,87 @@
+# Arm SiP service configuration options
+
+# Copyright (c) 2022, Intel Corporation. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig ARM_SIP_SVC_ARM64
+	bool "Support for the Arm SiP"
+	depends on ARM64
+	depends on HAS_ARM_SMCCC
+	help
+	  Enable Zephyr to communicate with secure system firmware
+	  implementing the platform-specifics via Arm SMC.
+
+if ARM_SIP_SVC_ARM64
+
+module = ARM_SIP_SVC_ARM64
+module-str = arm_sip_svc
+
+config ARM_SIP_SVC_HAS_HVC_CALLS
+	bool "Enables HVC calls to EL2 layer"
+	default n
+	depends on ARM_SIP_SVC_ARM64
+
+config ARM_SIP_SVC_HAS_SMC_CALLS
+	bool "Enables SMC calls to EL3 layer"
+	default n
+	depends on ARM_SIP_SVC_ARM64
```

Comment:
We should leverage the DT for this, not using Kconfig symbols. See for example `drivers/pm_cpu_ops/pm_cpu_ops_psci.c`. We already ship a `method` DT property that defines whether we are going to use HVC or SMC.

Reply:
Hi drivers/pm_cpu_ops/pm_cpu_ops_psci.c is a driver implementation.
To achieve the similar implementation here I would need to add a dts entry and yaml file ,but I am not sure if that is the correct way to go in a subsystem.
Also We had initially tried to upstream this as a driver , but was told  that the implementation is more suited as a subsystem. So we made the necessary changes and submitted it here
https://github.com/zephyrproject-rtos/zephyr/pull/43256#pullrequestreview-896090862.
Can you suggest how to go ahead for this?

Reply:
AFAIK, the sip_svc is designed as an interface right? Generally, the whole thing should consist of 2 parts: 1) the interface part which belongs to subsys, and 2) the implementation part which belongs to the driver, and implements the sip_svc subsys. Doing this is because we expect the sip_svc interface/subsys to have many implementations with different platforms.
So this HVC/SVC calls should be the driver part instead of subsys part.
Not sure if I understand right and if I make it clear?

Reply:
I have updated the PR, with implementation for driver subsystem split